### DISCUSSION TITLE
Offline installation of docToolchain v2.x

### DISCUSSION THREAD

**Discussion by pssp-sven** [docToolchain/docToolchain#685] [Q&A]
Hello,
do you have any recommendations or even docs on how to install docToolchain v2.x offline?
dtcw is a wonderful wrapper and mechanism to install docToolchain if you're online. But in my current project it's not possible to use it that way. I tried just to copy the folders ".doctoolchain" and ".gradle" from the user profile which works, but then there are still the maven dependencies missing. Maybe there is a better way to do it, that I'm not aware of.
Thanks and best regards

**Comment by rdmueller** [comment]
I ahve to admit that I was also looking for such a solution when I created the docker image.
I noticed that gradle, even when the dependencies where already downloaded, tries to refresh those.

It seems that `./dtcw generateSite --offline` could do the trick https://www.baeldung.com/gradle-offline-mode#:~:text=%20Gradle%20Offline%20Mode%20%201%20Overview.%20Gradle,IntelliJ%20IDEA%20and%20Eclipse%29%2C%20so%20we...%20More%20 .

Haven't tried it yet - it would be great if you could test it.

**Reply by pssp-sven** [reply]
Hi Ralf,

thanks for your advice! I'll keep it in mind, if gradle tries to access network resources that are already present on the system. But right now it's not necessary to use the `--offline` option, since I've downloaded and copied all dependencies to the target system beforehand that are needed for the tasks I'm already aware of. In that case everything works as expected. But still the user could try to use a task, that was not downloaded before.

So my question is more about how a complete offline installation could take place, which automatically brings all conceivable and currently usable dependencies with it.

Thanks again for your support.

**Reply by rdmueller** [reply]
I guess, even if you've downloaded all dependencies, gradle will try to check for updates after some days/weeks.
There are mechanisms to copy those dependencies to a flat file repository.
So, maybe, you need the `--offline` option in a few days :-)
